### PR TITLE
Fixed CancerType/CancerTypeDetails tables hang after interaction

### PR DESCRIPTION
**Pull request by victoria34**

### CODE REVIEW

**Review by zhx828 — CHANGES_REQUESTED** [review]
Please remove unnecessary variables and functions. Thanks.

File: app/scripts/views/components/dataTable/tableViewTemplate.js [hunk 1]
```diff
@@ -83,10 +84,12 @@
                 'patient_uid' : 'sample_uid';
             var _selectedCases =
               _.pluck(this.invisibleDimension.top(Infinity), attrId);
-            this.chartInst.update(_selectedCases, this.selectedRows);
-            this.setDisplayTitle(this.chartInst.getCases().length);
-            this.showLoad = false;
-            this.showRainbowSurvival();
+            if (this.dataLoaded) {
```
Comment (zhx828) [inline app/scripts/views/components/dataTable/tableViewTemplate.js]: This should be included with the else statement above

File: app/scripts/views/components/dataTable/tableViewTemplate.js [hunk 2]
```diff
@@ -203,11 +207,25 @@
...
+        
+        this.dataLoaded = true;
+        
         this.chartInst.init(this.attributes, opts, this.$root.selectedsampleUIDs,
           this.$root.selectedgenes, data, {
             addGeneClick: this.addGeneClick,
             submitClick: this.submitClick
           }, this.isMutatedGeneCna ? _data.geneMeta : null, this.invisibleDimension, this.genePanelMap);
+
+        if (this.selectedRows) {
```
Comment (zhx828) [inline app/scripts/views/components/dataTable/tableViewTemplate.js]: The condition should be decided based on `_selectedCases`

**Review by zhx828 — CHANGES_REQUESTED** [review]

File: app/scripts/views/components/dataTable/tableViewTemplate.js (hunk 2 above)
Comment (victoria34) [reply app/scripts/views/components/dataTable/tableViewTemplate.js]: I have changed it to if (_selectedCases.length > 0). Thanks!

File: app/scripts/views/components/dataTable/tableViewTemplate.js (hunk 1 above)
Comment (victoria34) [reply app/scripts/views/components/dataTable/tableViewTemplate.js]: if (this.madeSelection && !this.isMutatedGeneCna) {
            this.madeSelection = false;
          } else if (this.dataLoaded) {
            var attrId =
              this.attributes.group_type === 'patient' ?
                'patient_uid' : 'sample_uid';
            var _selectedCases =
              _.pluck(this.invisibleDimension.top(Infinity), attrId);
            this.chartInst.update(_selectedCases, this.selectedRows);
            this.setDisplayTitle(this.chartInst.getCases().length);
            this.showLoad = false;
            this.showRainbowSurvival();
          }

**Review by zhx828 — APPROVED** [review]

**Comment by zhx828** [comment]
This closes #445

**Pull request merged**